src/modules/Core/Wallets/api.js

```diff
@@ -36,72 +32,43 @@ const dummyAbcReceiveAddress: AbcReceiveAddress = {
 }
 
-export const setTransactionDetailsRequest = (wallet: AbcCurrencyWallet, txid: string, currencyCode: string, abcMetadata: AbcMetadata): Promise<void> =>
-  // console.log('wallet is: ', wallet)
-  // console.log('currencyCode is: ', currencyCode)
-  // console.log('transactionDetails: ', transactionDetails)
-  //  parameters should be txid, currencyCode, and then metaData
-   wallet.saveTxMetadata ? wallet.saveTxMetadata(txid, currencyCode, abcMetadata): Promise.resolve()
+export const setTransactionDetailsRequest = (wallet: AbcCurrencyWallet, txid: string, currencyCode: string, abcMetadata: AbcMetadata): Promise<void> => {
+  return wallet.saveTxMetadata ? wallet.saveTxMetadata(txid, currencyCode, abcMetadata): Promise.resolve()
+}
 
 export const getReceiveAddress = (wallet: AbcCurrencyWallet, currencyCode: string): Promise<AbcReceiveAddress> => {
-  if (wallet.getReceiveAddress) {
-    return wallet.getReceiveAddress(currencyCode)
-  } else {
-    return Promise.resolve(dummyAbcReceiveAddress)
-  }
+  return wallet.getReceiveAddress ? wallet.getReceiveAddress(currencyCode) : Promise.resolve(dummyAbcReceiveAddress)
 }
 
 export const makeSpend = (wallet: AbcCurrencyWallet, spendInfo: AbcSpendInfo): Promise<AbcTransaction> => {
-  // console.log('spendInfo', spendInfo)
-  if (wallet.makeSpend) {
-    return wallet.makeSpend(spendInfo)
-  } else {
-    return Promise.resolve(dummyAbcTransaction)
-  }
+  return wallet.makeSpend ? wallet.makeSpend(spendInfo) : Promise.resolve(dummyAbcTransaction)
 }
 
 export const getMaxSpendable = (wallet: AbcCurrencyWallet, spendInfo: AbcSpendInfo): Promise<string> => {
-  if (wallet.getMaxSpendable) {
-    return wallet.getMaxSpendable(spendInfo)
-  } else {
-    return Promise.resolve('0')
-  }
+  return wallet.getMaxSpendable ? wallet.getMaxSpendable(spendInfo) : Promise.resolve('0')
 }
 
 export const getBalance = (wallet: AbcCurrencyWallet, currencyCode: string): string => {
-  if (wallet.getBalance) {
-    return wallet.getBalance({currencyCode})
-  } else {
-    return '0'
-  }
+  return wallet.getBalance ? wallet.getBalance({currencyCode}) : '0'
 }
 
 export const disableTokens = (wallet: AbcCurrencyWallet, tokens: Array<string>) => {
   return wallet.disableTokens(tokens)
 }
 
-
 export const enableTokens = (wallet: AbcCurrencyWallet, tokens: Array<string>) => {
   return wallet.enableTokens(tokens)
 }
 
 export const addCoreCustomToken = (wallet: AbcCurrencyWallet, tokenObj: any) => {
   return wallet.addCustomToken(tokenObj)
-  .then(() => {
-    wallet.enableTokens([tokenObj.currencyCode])
-    .then(() => {
-      return
-    })
-    .catch((e) => console.log(e))
-  })
+  .then(() => wallet.enableTokens([tokenObj.currencyCode]))
   .catch((e) => console.log(e))
 }
 
 export const getEnabledTokensFromFile = (wallet: AbcCurrencyWallet): Promise<Array<any>> => {
   return getEnabledTokensFile(wallet).getText()
-  .then((text) => {
-    return JSON.parse(text)
-  })
+  .then(JSON.parse)
```

Comment:
Will this change not make it harder to debug? No way of checking the returned text before the code attempts to parse it?

Reply:
yeah, ill undo this one, i just like the way it looks better